### ISSUE TITLE
[FR]

### ISSUE THREAD

**Issue by Dhaval08** [mlflow/mlflow#3222]
Thank you for submitting a feature request. **Before proceeding, please review MLflow's [Issue Policy for feature requests](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md#feature-requests) and the [MLflow Contributing Guide](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst)**.

## Proposal Summary

(In a few sentences, provide a clear, high-level description of the feature request)
Log streaming for ongoing runs. A mechanism to view the logs of an ongoing run on the MLFlow UI.

## Motivation
For use cases where users don't have access to the cloud node on which the experiment is running, this would be a helpful feature to view the logs and debug issues if any.

**Comment by dmatrix** [comment]
@Dhaval08 thanks for filing the feature enhancement. What would you prefer: live tail of the logs in the UI or after the run to inspect the logs?

**Comment by Dhaval08** [comment]
@dmatrix  Live tail of logs on the UI. So that the users can keep track of an ongoing job and even debug in case the job fails. At the moment, if the job fails, there's no way to debug the issue if you don't have access to the cloud node that's running the job.

**Comment by Dhaval08** [comment]
@dmatrix any update on this?
Thank you.

**Comment by dmatrix** [comment]
@Dhaval08 I wanted to circle back and clarify some functionality delineation that might be misconstrued. Or we elected the wrong phrases in the documentation that might have easily misinterpreted.

MLflow does _not manage_ your machine learning training "runs." It only records your machine learning runs' results: metrics, parameters, artifacts (models, gifs, text files, etc). The client, your machine learning code instruments these as API calls to the MLflow Tracking Service, using the MLflow Fluent APIs ([mlflow](https://mlflow.org/docs/latest/python_api/mlflow.html) and [mlflow.tracking](https://mlflow.org/docs/latest/python_api/mlflow.tracking.html)). These at the low-level get converted to REST calls to the MLflow Tracking Service. 

It there's an error, your client code will get an exception, which you can handle however you want. For live training run, for example, 
if you are training a PyTorch or TensorFlow/Keras model, all the epoch and related metrics are printed by the respective library in your client to stdout. By the client here, I mean, your Python code using these ML libraries.

MLflow UI or Tracking Serice has no handle or access to your client code per se. The MLflow tracking server will log any error to stdout and stderr, but those might have nothing to do with your training code. 

In short, MLFlow does not manage your runs or has access to your training code while your ML code is running and training your model. Rather it manages or records the results or outcome of your runs if you instrument them as such.

Therefore, this feature request, as I understand what you want live log in the MLflow UI of your training code, is out of its scope of functionality. 

Let me know if you have any further questions, or I misunderstood your ask or request. 
Cheers

**Comment by jcuomo-asapp** [comment]
Being able to stream logs in real time is useful because in those cases where the instance you are running your training fails on Out of Memory you cannot handle that exception. For example weights and biases does that automatically.

**Comment by GeorgePearse** [comment]
Bump, would also be a fan of this, both Aim and WnB include it.

**Comment by rutpoule** [comment]
I'd like to suggest the alternative functionality suggested by @dmatrix 
I found it pretty straightforward to use python logging to write the log file into the artifacts directory for a run.  It would be extremely useful to have a log inspector in the web UI to highlight and filter messages by severity and search the text.  This capability would be very helpful for debugging runs.

**Comment by SebastianScherer88** [comment]
Another approach could be for mlflow to implement a `TrackingStreamHandler` logging handler class that users coyld easily use to send logs in real time to tge mlflow tracking server

**Comment by GeorgePearse** [comment]
@dmatrix almost every alternative to MLFlow includes it, so it seems very much within the potential "scope of functionality"

**Comment by asiron** [comment]
I completely agree that if it's not included automatically like in WandB, where all of your stdout is captured and streamed in real-time to the tracking server. However, it should be at least possible to do it with minimal effort if your using python logging library
